### PR TITLE
[M] CANDLEPIN-819: Added max page size option for request paging

### PR DESCRIPTION
**Pull request by Ceiu**
- Added a new option, candlepin.paging.max_page_size, for controlling the maximum size of a given page request; defaults to 3000
- Updated the PageRequestFilter to throw a BadRequestException if a page size is specified larger than the configured max page size

### CODE REVIEW

File: src/main/java/org/candlepin/config/ConfigProperties.java [hunk 1]
```diff
@@ -293,6 +293,10 @@ private ConfigProperties() {
     // How long (in seconds) to wait for job threads to finish during a graceful Tomcat shutdown
     public static final String ASYNC_JOBS_THREAD_SHUTDOWN_TIMEOUT = "candlepin.async.thread.shutdown.timeout";
 
+    // The maximum allowable size for pages; defaults to 3000
+    public static final String PAGING_MAX_PAGE_SIZE = "candlepin.paging.max_page_size";
```
Comment (nikosmoum) [inline src/main/java/org/candlepin/config/ConfigProperties.java:297]: There's already a config property like this, the suggestion was to update its default value

**Review by nikosmoum — COMMENTED** [review]
this change should go in the feature/candlepinquery_removal feature branch

File: src/main/java/org/candlepin/config/ConfigProperties.java (hunk 1 above)
Comment (Ceiu) [reply src/main/java/org/candlepin/config/ConfigProperties.java:297]: Oh good, I thought it was kinda odd I didn't see them when I searched for them. I must have typo'd "paging" while looking for the existing ones.

Fixed.

**Comment by Ceiu** [comment]
@nikosmoum Going to close this PR and reopen it with a new branch name against the target branch.

**Pull request closed without merging**